CoffeeChat/Cells/OnboardingTableViewCell.swift

```diff
@@ -77,55 +85,53 @@ class OnboardingTableViewCell: UITableViewCell {
-            if showingInterests {
+            make.trailing.equalToSuperview().inset(12)
+            if categoriesLabel.text != nil {
                 make.top.equalToSuperview().inset(8.5)
             } else {
                 make.centerY.equalToSuperview()
             }
         }
 
-        if showingInterests {
-            categoriesLabel.snp.remakeConstraints { make in
-                make.leading.equalTo(titleLabel)
-                make.top.equalTo(titleLabel.snp.bottom)
-            }
+        categoriesLabel.isHidden = categoriesLabel.text == nil
+        categoriesLabel.snp.remakeConstraints { make in
+             make.leading.equalTo(titleLabel)
+             make.top.equalTo(titleLabel.snp.bottom)
         }
     }
 
     func configure(with interest: Interest) {
         titleLabel.text = interest.name
-        categoriesLabel.text = interest.categories
-        interestImageView.image = UIImage(named: interest.image)
-        // Determine if a relayout is needed (was showing Group and configured with an Interest)
-        if !initialized || !showingInterests {
-            relayoutSubviews(interests: true)
+        categoriesLabel.text = interest.categories?.joined(separator: ", ") ?? nil
+        if let pictureString = interest.imageURL {
+            interestImageView.kf.setImage(with: URL(string: pictureString))
+        } else {
+            interestImageView.image = nil
+        }
+
+        if lastShownItem != .interest {
+            lastShownItem = .interest
+            setupConstraints()
         }
     }
 
     func configure(with group: Group) {
         titleLabel.text = group.name
-        // Determine if a relayout is needed (was showing Interest and configured with an Group)
-        if !initialized || showingInterests {
-            relayoutSubviews(interests: false)
+        categoriesLabel.text = nil
+        if let pictureString = group.imageURL {
+            interestImageView.kf.setImage(with: URL(string: pictureString))
+        } else {
+            interestImageView.image = nil
         }
-    }
-
-    func changeColor(isSelected: Bool) {
-        backdropView.backgroundColor = isSelected ? .pearGreen : .white
-    }
 
-    /// Removes and resets constraints on the cell. Includes categoriesLabel in the layout if `interests` is true
-    private func relayoutSubviews(interests: Bool) {
-        showingInterests = interests
-        initialized = true
-        setupConstraints(showingInterests: showingInterests)
+        if lastShownItem != .group {
+            lastShownItem = .group
+            setupConstraints()
+        }
     }
 
-    override func setSelected(_ selected: Bool, animated: Bool) {
-        super.setSelected(selected, animated: animated)
-        if shouldSelectionChangeAppearence {
-            changeColor(isSelected: isSelected)
-        }
+    func changeColor(selected: Bool) {
+        backdropView.backgroundColor = selected ? .pearGreen : .white
     }
```

Comment:
Is there any reason not to in-line this above? If you want to keep it like this, I think it could be private.

Reply:
This function is called by `EditingViewController` to handle the highlight of the cells for the different sections